contracts/AmbireAccount.sol

```diff
@@ -140,7 +140,12 @@ contract AmbireAccount {
 			require(privileges[signerKey] != bytes32(0), 'INSUFFICIENT_PRIVILEGE');
 		}
 
+		// we increment the nonce to prevent reentrancy
+		// also, we do it here as we want to reuse the previous nonce
+		// and respectively hash upon recovery / canceling
+		nonce = currentNonce + 1;
```

Comment:
you can add a comment that doing this after sig verification is fine because sig verification can only do STATICCALLS